gnomad_qc/v3/create_release/sanity_checks.py

```diff
@@ -0,0 +1,711 @@
+    :return: None
+    :rtype: None
+    """
+    t = t.rows() if isinstance(t, hl.MatrixTable) else t
+
+    for subset in subsets:
+        if subset != "":
+            subset += delimiter
+        else:
+            continue
+        for subfield in ["AC", "AN", "nhomalt"]:
+            logger.info("adj checks")
+            subfield_label = f"{subfield}{delimiter}adj"
+            subfield_subset_label = f"{subfield}{delimiter}{subset}adj"
+
+            generic_field_check(
+                t,
+                cond_expr=(t.info[subfield_label] == t.info[subfield_subset_label]),
+                check_description=f"{subfield_label} != {subfield_subset_label}",
+                display_fields=[
+                    f"info.{subfield_label}",
+                    f"info.{subfield_subset_label}",
+                ],
+                verbose=verbose,
+                show_percent_sites=show_percentage_sites,
+            )
+
+    freq_counts = t.aggregate(
+        hl.struct(
+            total_defined_AC=hl.agg.count_where(hl.is_defined(t.info["AC-adj"])),
+            total_defined_AC_raw=hl.agg.count_where(hl.is_defined(t.info["AC-raw"])),
+        )
+    )
+    logger.info(f"Frequency spot check counts: {freq_counts}")
+
+
+def sample_sum_check(
+    t: Union[hl.MatrixTable, hl.Table],
+    subset: str,
+    label_groups: Dict[str, List[str]],
+    verbose: bool,
+    subpop: bool = None,
+    sort_order: List[str] = SORT_ORDER,
+    delimiter: str = "-",
+) -> None:
+    """
+    Compute afresh the sum of call stats annotations for a specified group of annotations, and compare to the annotated version;
+    show the results from checking the sum of the specified annotations in the terminal.
+    :param hl.MatrixTable, hl.Table t: Input MatrixTable or Table containing annotations to be summed.
+    :param subset: String indicating sample subset.
+    :param label_groups: Dictionary containing an entry for each label group, where key is the name of the grouping,
+        e.g. "sex" or "pop", and value is a list of all possible values for that grouping (e.g. ["male", "female"] or ["afr", "nfe", "amr"]).
+    :param verbose: If True, show top values of annotations being checked, including checks that pass; if False,
+        show only top values of annotations that fail checks.
+    :param subpop: Subpop abbreviation, supplied only if subpopulations are included in the annotation groups being checked.
+    :param sort_order: List containing order to sort label group combinations. Default is SORT_ORDER.
+    :param delimiter: String to use as delimiter when making group label combinations.
+    :return: None
+    """
+
+    t = t.rows() if isinstance(t, hl.MatrixTable) else t
+
+    if subset != "":
+        subset += delimiter
+
+    label_combos = make_label_combos(label_groups, label_delimiter=delimiter)
+    combo_AC = [t.info[f"AC-{subset}{x}"] for x in label_combos]
```

Comment:
should this `-` be based on `delimiter` and same question for the two below? Or is there a need for another delimiter for this?